### PR TITLE
feat(devmanual): Add \OCP\DB\IResult::fetch deprecation

### PR DESCRIPTION
**Pull request by ChristophWurst**
### ☑️ Resolves

* Companion of https://github.com/nextcloud/server/pull/40655

### Changes

* Adjust database query builder usage
* Document deprecation and replacements

### 🖼️ Screenshots

image

### CODE REVIEW

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.rst [hunk 1]
```diff
@@ -71,6 +71,7 @@ Changed APIs
 Deprecated APIs
 ^^^^^^^^^^^^^^^
 
+* ``\OCP\DB\IResult::fetch``: use the new ``fetchAssociative``, ``fetchNumeric`` and ``fetchOne`` instead.
```
Comment (nickvergessen) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.rst:74]: Maybe we can help with outlining which one was the old behaviour? :)

**Review by blizzz — APPROVED** [review]

**Review by fsamapoor — APPROVED** [review]

**Comment by juliushaertl** [comment]
Mind to squash?

**Review by juliushaertl — APPROVED** [review]

**Comment by blizzz** [comment]
rebased and squashed

**Pull request merged**